### PR TITLE
Add Zipformer recipe for GigaSpeech

### PR DESCRIPTION
**Pull request by yfyeung**
- [x] Add Zipformer recipe for GigaSpeech
- [x] Upload checkpoints, exported models, logs, and bpe models trained using the GigaSpeech XL dataset to https://huggingface.co/yfyeung/icefall-asr-gigaspeech-zipformer-2023-10-17
- [x] Update https://github.com/k2-fsa/icefall/blob/master/egs/gigaspeech/ASR/RESULTS.md
- [x] Update https://github.com/k2-fsa/icefall/blob/master/README.md
- [x] Update https://github.com/yfyeung/icefall/blob/phone2/egs/gigaspeech/ASR/README.md

### CODE REVIEW

**Comment by desh2608** [comment]
@yfyeung do you have a pretrained model you could share?

**Comment by yfyeung** [comment]
> @yfyeung do you have a pretrained model you could share?

Only M subset.

**Comment by yfyeung** [comment]
> @yfyeung do you have a pretrained model you could share?

Sure, check https://huggingface.co/yfyeung/icefall-asr-gigaspeech-zipformer-2023-10-17

File: egs/gigaspeech/ASR/zipformer/asr_datamodule.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
+# Copyright      2021  Piotr Żelasko
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/asr_datamodule.py:1]: Is it possible to replace it with a symlnik?

File: egs/gigaspeech/ASR/zipformer/export-onnx-ctc.py [hunk 2]
```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/export-onnx-ctc.py:1]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/export-onnx-streaming.py [hunk 3]
```diff
@@ -0,0 +1,775 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/export-onnx-streaming.py:1]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/export.py [hunk 4]
```diff
@@ -0,0 +1,522 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/export.py:1]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/jit_pretrained.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python3
+# Copyright 2021-2023 Xiaomi Corporation (Author: Fangjun Kuang, Zengwei Yao)
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/jit_pretrained.py:2]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/jit_pretrained_ctc.py [hunk 6]
```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/jit_pretrained_ctc.py:1]: can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/jit_pretrained_streaming.py [hunk 7]
```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/jit_pretrained_streaming.py:1]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/zipformer/onnx_check.py [hunk 8]
```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/zipformer/onnx_check.py:1]: Can you replace it with a symlink?

File: egs/gigaspeech/ASR/RESULTS.md [hunk 9]
```diff
@@ -1,4 +1,75 @@
...
+
+See <https://github.com/k2-fsa/icefall/pull/1254> for more details.
+
+[zipformer](./zipformer)
+
+- Non-streaming
+- normal-scaled model, number of model parameters: 65549011, i.e., 65.55 M
+
+You can find a pretrained model, training logs, decoding logs, and decoding results at:
+<https://huggingface.co/yfyeung/icefall-asr-gigaspeech-zipformer-2023-10-17>
```
Comment (csukuangfj) [inline egs/gigaspeech/ASR/RESULTS.md:12]: Could you upload the tensorboard log to
https://wandb.ai/site
and post a link to it here?

**Review by csukuangfj — DISMISSED** [review]
Thanks! 

The PR gives the state-of-the-art WER for gigapseech in icefall!

Left some minor comments.

File: README.md [hunk 10]
```diff
@@ -148,8 +148,11 @@ in the decoding.
 
 ### GigaSpeech
 
-We provide two models for this recipe: [Conformer CTC model][GigaSpeech_conformer_ctc]
-and [Pruned stateless RNN-T: Conformer encoder + Embedding decoder + k2 pruned RNN-T loss][GigaSpeech_pruned_transducer_stateless2].
+We provide three models for this recipe: [Zipformer]
```
[the remark this answers is not in the document]
Comment (yfyeung) [reply README.md]: OK, I removed it. It's redundant.

File: egs/gigaspeech/ASR/zipformer/asr_datamodule.py (hunk 1 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/asr_datamodule.py:1]: I have modified this file. In the past, GigaSpeech XL splits would be merged. This one uses `lhotse.mux`.

File: egs/gigaspeech/ASR/zipformer/export-onnx-ctc.py (hunk 2 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/export-onnx-ctc.py:1]: Sure.

File: egs/gigaspeech/ASR/zipformer/export-onnx-streaming.py (hunk 3 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/export-onnx-streaming.py:1]: Sure.

File: egs/gigaspeech/ASR/zipformer/export.py (hunk 4 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/export.py:1]: I changed some comments like huggingface link in it.

File: egs/gigaspeech/ASR/zipformer/jit_pretrained.py (hunk 5 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/jit_pretrained.py:2]: Sure.

File: egs/gigaspeech/ASR/zipformer/jit_pretrained_streaming.py (hunk 7 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/jit_pretrained_streaming.py:1]: Sure.

File: egs/gigaspeech/ASR/zipformer/onnx_check.py (hunk 8 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/onnx_check.py:1]: Sure.

File: egs/gigaspeech/ASR/RESULTS.md (hunk 9 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/RESULTS.md:12]: OK.

File: egs/gigaspeech/ASR/zipformer/jit_pretrained_ctc.py (hunk 6 above)
Comment (yfyeung) [reply egs/gigaspeech/ASR/zipformer/jit_pretrained_ctc.py]: Sure

File: egs/gigaspeech/ASR/RESULTS.md (hunk 9 above)
Comment (csukuangfj) [inline egs/gigaspeech/ASR/RESULTS.md:12]: Could you add a CI test for your model?

Comment (csukuangfj) [reply egs/gigaspeech/ASR/RESULTS.md:12]: By the way, the link has not been posted. please leave a message when you think this PR is ready to merge.

Comment (yfyeung) [reply egs/gigaspeech/ASR/RESULTS.md:12]: https://wandb.ai/yifanyeung/icefall-asr-gigaspeech-zipformer-2023-10-20

Comment (yfyeung) [reply egs/gigaspeech/ASR/RESULTS.md:12]: OK, I have posted that.

Comment (yfyeung) [reply egs/gigaspeech/ASR/RESULTS.md:12]: Ok, I have added.

**Comment by yfyeung** [comment]
Please re-review this PR. @csukuangfj

**Review by csukuangfj — APPROVED** [review]
Thanks!

**Comment by yuyun2000** [comment]
I want a brand new streaming English model

**Pull request merged**